spec/models/change_service_event_spec.rb

```diff
@@ -21,6 +21,22 @@
       expect(change_service_event.data['name']).to eql(new_service_name)
     end
   end
+  context '#trigger_publish_event' do
+    it 'when change service name publish is trigger' do
+      new_entity_id = 'https://changed-entity-id'
+      service.assign_attributes(entity_id: new_entity_id)
+      change_service_event = create(:change_service_event, service: service)
+      component_1 = create(:sp_component)
+      event = create(:assign_sp_component_to_service_event, service: service, sp_component_id: component_1.id)
+      update_entity_id = 'https://changed-entity-id1'
+      service.assign_attributes(entity_id: update_entity_id)
+      change_service_event = create(:change_service_event, service: service)
+
+      resulting_event = PublishServicesMetadataEvent.last
+      expect(resulting_event).to be_present
+      expect(resulting_event.data['services_metadata']['connected_services'][0]['entity_id']).to eq(update_entity_id)
+    end
+  end
   context 'failure' do
```

Comment:
you should also have a test for MSA change and a test when the service shouldn't trigger the publishevent

Reply:
unresolving this as it hasn't been adressed along with https://github.com/alphagov/verify-self-service/pull/373#discussion_r392156578